pkg/platform/kube/controller/nucliofunction.go

```diff
@@ -551,3 +555,34 @@ func (fo *functionOperator) enrichNodeSelector(ctx context.Context, function *nu
 		"NodeSelector", function.Status.EnrichedNodeSelector)
 	return nil
 }
+
+func (fo *functionOperator) enrichAndValidateServiceAccount(ctx context.Context, function *nuclioio.NuclioFunction) error {
+	projectName, ok := function.Labels[common.NuclioResourceLabelKeyProjectName]
+	if !ok {
+		return errors.New("Function does not have a project label, cannot validate service account")
+	}
+
+	if !fo.controller.platformConfiguration.Kube.IsConfiguredToEnrichServiceAccountFromProject() &&
+		!fo.controller.platformConfiguration.Kube.IsConfiguredToVerifyServiceAccountFromProject() {
```

Comment:
What if only one of them is configured? enrich, no validate / validate, no enrich

Reply:
@TomerShor then run enrich and validate and it will stop there earlier as well